File: chazm-model/src/main/java/runtimemodels/chazm/model/event/SpecificationGoalEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.SpecificationGoal;

/**
 * The {@linkplain SpecificationGoalEventFactory} interface defines the API for constructing {@linkplain SpecificationGoalEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface SpecificationGoalEventFactory {

    /**
     * Constructs an {@linkplain SpecificationGoalEvent}.
     *
     * @param category the {@linkplain EventCategory}.
     * @param goal     the {@linkplain SpecificationGoal}.
     * @return a {@linkplain SpecificationGoalEvent}.
     */
    SpecificationGoalEvent build(EventCategory category, SpecificationGoal goal);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/SpecificationGoalEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.SpecificationGoal

/**
 * The [SpecificationGoalEventFactory] interface defines the API for constructing [SpecificationGoalEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface SpecificationGoalEventFactory {

    /**
     * Constructs an [SpecificationGoalEvent].
     *
     * @param category the [EventType].
     * @param goal     the [SpecificationGoal].
     * @return a [SpecificationGoalEvent].
     */
    fun build(category: EventType, goal: SpecificationGoal): SpecificationGoalEvent

}
